vpr/src/base/atom_netlist.cpp

```diff
@@ -102,6 +102,18 @@ AtomBlockId AtomNetlist::find_atom_pin_driver(const AtomBlockId blk_id, const t_
     return AtomBlockId::INVALID();
 }
 
+std::string AtomNetlist::get_assigned_net_name(const std::string alias_net_name) const {
+    std::string assigned_net_name;
+
+    auto result = clock_net_names_map_.find(alias_net_name);
+
+    if (result != clock_net_names_map_.end()) {
+        assigned_net_name = result->second;
+    }
+
+    return assigned_net_name;
```

Comment:
It looks like this will return the empty string for a net name which is not in the alias list?

This should check for an alias first (via `clock_net_names_map_`), using the re-mapped name if found or otherwise the passed argument (`alias_net_name`). It should then check that the 'resolved' net name is actually valid (to handle the case of an invalid `alias_net_name` being passed).

Reply:
One of the original reasons I did leave the empty string is that it is possible that the SDC target clock has a wildcard in the name.
If this happens, the alias map would not contain the alias net, and I didn't think it was a good idea to output the alias name in case the assigned net name was not found.

What I am doing now is to check if the net alias is a valid clock net, and, if not, it returns an empty string.

Reply:
> One of the original reasons I did leave the empty string is that it is possible that the SDC target clock has a wildcard in the name.

Hmm, I see your point.

I think it would be best to leave any wildcard handling to the SDC processing rather than the netlist data structures (which are more fundamental).

I'd suggest we do the following:
1. Add a method to the netlist which allows iterating through the net alias names (e.g. `AtomNetlist::net_aliases(AtomNetId net_id)`).
2. Have the SDC code iterate through both the current netlist net names (i.e. `AtomNetlist::nets()`), and the aliases (`AtomNetlist::net_aliases()`) to do wild-card name matching (and thereby capturing both current and aliased net names).
3. Have the SDC code resolve the wild card matches back to the actual netlist AtomNetId (via  `AtomNetlist::get_assigned_net_name()` and `AtomNetlist::find_net()`) to which the constraints actually apply.

Reply:
I have done something similar:

1. `AtomNetlist::net_aliases(AtomNetId)` which return the set of aliases for the specified net.
2. during buffer lut removal add both input net and output net to the aliases (also for the IPAD and OPAD related nets)
3. During SDC, for each net, iterate through the set of aliases (which contains also the assigned net name during buffer_removal) and check the regex match.

If buffer lut removal is disabled, the `net_aliases` function returns a set with the original net name. 

Each setter and getter for the aliases set has an assertion to verify the net is valid.